### PR TITLE
CI improvements

### PR DESCRIPTION
**Pull request by ffromani**
this script is kni-specific, hence belongs on hack-kni.
It was originally added on hack/ because it predated the hack/hack-kni
split.

Improve the verify-commits.sh script to be more precise in checking (just check the subject line) and to have better reporting
Fix the `unit-test-quick.sh` to be consistent with go.mod

Signed-off-by: Francesco Romani <fromani@redhat.com>

### CODE REVIEW

File: hack-kni/verify-commits.sh [hunk 1]
```diff
@@ -48,7 +48,7 @@ git log --oneline "$latest_upstream_commit"..HEAD
 echo
 
 # list commits
-for commitish in $(git log --oneline --no-merges "$latest_upstream_commit"..HEAD | cut -d' ' -f 1); do
+for commitish in $(git log --oneline --no-merges "$latest_upstream_commit"..HEAD~ | cut -d' ' -f 1); do
```
Comment (Tal-or) [inline hack-kni/verify-commits.sh]: The ```--no-merges``` flag should take care of that, so ```HEAD~``` won't cause skipping the check of the last commit?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -14,10 +14,5 @@ jobs:
         with:
           fetch-depth: 0
 
-      - name: set up golang
```
Comment (Tal-or) [inline .github/workflows/ci.yaml:17]: The commit is missing the [KNI] prefix but yet the CI passed. I really need to revisit the script

File: hack-kni/verify-commits.sh (hunk 1 above)
Comment (ffromani) [reply hack-kni/verify-commits.sh]: yes, this is just wrong. The fix (if needed) is elsewhere

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (ffromani) [reply .github/workflows/ci.yaml:17]: yup, this is what I was testing.

**Comment by Tal-or** [comment]
I think the problem is here:
https://github.com/openshift-kni/scheduler-plugins/blob/master/.github/hooks/commit-msg#L6
please change the line to:
``` if [[ "$(head -1 $cmtmsg)" != \[KNI\]* ]]; then```

I recall I fixed that when it was under my org, but maybe I forgot to push it or something

**Comment by Tal-or** [comment]
@fromanirh for the ci/prow/ci-unit the fix is: https://github.com/openshift-kni/scheduler-plugins/pull/10

**Comment by ffromani** [comment]
> @fromanirh for the ci/prow/ci-unit the fix is: #10

thanks, actually not sure we should use the go module name or the relative path.
Going with go module name mostly because this is what `hack/unit-test.sh` is doing.

**Pull request merged**